test/cpp_api_parity/functional_impl_check.py

```diff
@@ -0,0 +1,232 @@
+# The purpose of this test is to check that we have implementation parity between
+# a Python `torch.nn.functional` function and its corresponding C++ `torch::nn::functional`
+# function. Concretely, this test does the following:
+#
+# 1. Get a test params dict from common_nn.py, run forward pass on the Python functional
+# created using the test params.
+#
+# 2. Serialize the Python functional's forward input arguments, deserialize them
+# in C++ and use them as input for the C++ functional's forward pass.
+#
+# 3. Run the forward pass on the C++ functional, and serialize the C++ functional's
+# forward output.
+#
+# 4. Compare Python/C++ functional's forward output. If they are the same, then we
```

Comment:
Why do we need to check backwards in the NN parity test, but not in the functional parity test?

Reply:
Thanks a lot for the catch! Backwards in functional parity test should also be checked, and I will add it to the list of follow-up items to do.